packages/amplify-provider-awscloudformation/src/admin-modelgen.ts

```diff
@@ -25,88 +23,99 @@ export const adminModelgen = async (context: $TSContext, resources: $TSAny[]): P
-      2 * 1000,
-      2000 * 1000,
-      amplifyBackendInstance.amplifyBackend,
-    );
-    if (jobCompletionDetails.Status === 'COMPLETED') {
-      spinner.succeed('Successfully generated models in the cloud.');
-    } else {
-      throw new AmplifyError('ModelgenError', { message: `Failed to generate models in the cloud.` });
-    }
-  } catch (e) {
-    spinner.stop();
-    printer.error(`Failed to create models in the cloud: ${e.message}`);
+    // overwrite project config with config that forces codegen to output js to a temp location
+    stateManager.setProjectConfig(undefined, forceJSCodegenProjectConfig);
+
+    // generateModels and generateModelIntrospection print confusing and duplicate output when executing these codegen paths
+    // so pipe stdout to a file and then reset it at the end to suppress this output
+    await fs.ensureDir(absoluteTempOutputDir);
+    const tempStdoutWrite = fs.createWriteStream(path.join(absoluteTempOutputDir, 'temp-console-log.txt'));
+    process.stdout.write = tempStdoutWrite.write.bind(tempStdoutWrite);
+
+    // invokes https://github.com/aws-amplify/amplify-codegen/blob/main/packages/amplify-codegen/src/commands/models.js#L60
+    await context.amplify.invokePluginMethod(context, 'codegen', undefined, 'generateModels', [context]);
+
+    // generateModelIntrospection expects --output-dir option to be set
+    _.set(context, ['parameters', 'options', 'output-dir'], relativeTempOutputDir);
+
+    // invokes https://github.com/aws-amplify/amplify-codegen/blob/main/packages/amplify-codegen/src/commands/model-intropection.js#L8
+    await context.amplify.invokePluginMethod(context, 'codegen', undefined, 'generateModelIntrospection', [context]);
+
+    const localSchemaPath = path.join(pathManager.getResourceDirectoryPath(undefined, 'api', resourceName), 'schema.graphql');
+    const localSchemaJsPath = path.join(absoluteTempOutputDir, 'models', 'schema.js');
+    const localModelIntrospectionPath = path.join(absoluteTempOutputDir, 'model-introspection.json');
+
+    // ==================== DO NOT MODIFY THIS MAP UNLESS YOU ARE 100% SURE OF THE IMPLICATIONS ====================
+    // this map represents an implicit interface between the CLI and the Studio CMS frontend
+    const s3ApiModelsPrefix = `models/${resourceName}/`;
+    const cmsArtifactLocalToS3KeyMap: Record<LocalPath, S3Key> = {
+      [localSchemaPath]: `${s3ApiModelsPrefix}schema.graphql`,
+      [localSchemaJsPath]: `${s3ApiModelsPrefix}schema.js`,
+      [localModelIntrospectionPath]: `${s3ApiModelsPrefix}modelIntrospection.json`,
+    };
+    // ==================== DO NOT MODIFY THIS MAP UNLESS YOU ARE 100% SURE OF THE IMPLICATIONS ====================
+
+    await uploadCMSArtifacts(await S3.getInstance(context), cmsArtifactLocalToS3KeyMap);
+  } finally {
+    process.stdout.write = originalStdoutWrite;
+    await fs.remove(absoluteTempOutputDir);
+    stateManager.setProjectConfig(undefined, originalProjectConfig);
```

Comment:
If removing the temp file fails and throws, the original config could be lost.

Reply:
Good call, updated resetting the project config to be the first line of the finally block